src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -277,7 +277,7 @@ protected function compileRawEchos($value)
 		{
 			$whitespace = empty($matches[3]) ? '' : $matches[3].$matches[3];
 
-			return $matches[1] ? substr($matches[0], 1) : '<?php echo '.$this->compileEchoDefaults($matches[2]).'; ?>'.$whitespace;
+			return $matches[1] ? substr($matches[0], 1) : '<?= '.$this->compileEchoDefaults($matches[2]).'; ?>'.$whitespace;
```

Comment:
You're wrong. `<?=` is not even considered as a short tag anymore, and is not affected by `short_open_tag`. See http://svn.php.net/viewvc?view=revision&revision=311260, from Rasmus Lerdorf himself.